src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -617,7 +618,7 @@ private async ValueTask<(Socket, Stream, TransportContext, HttpResponseMessage)>
                     case HttpConnectionKind.ProxyTunnel:
                     case HttpConnectionKind.SslProxyTunnel:
                         HttpResponseMessage response;
-                        (stream, response) = await EstablishProxyTunnel(cancellationToken).ConfigureAwait(false);
+                        (stream, response) = await EstablishProxyTunnel(request.Headers, cancellationToken).ConfigureAwait(false);
```

Comment:
`request.Headers` => `request.HasHeaders ? request.Headers : null` in order to avoid unnecessarily lazily initializing the headers collection